File: CompetitiveProgramming/CCC/CCC '04 J4 - Simple Encryption/2019-04-15/SimpleEncryption.cpp
#include <iostream>
using namespace std;

int main()
{
    cin.sync_with_stdio(0);
    cin.tie(0);
	
	string top, line;
	getline(cin, top);
	getline(cin, line);
	
	int shift[top.length()];
	for(int i = 0; i < top.length(); i++)
		shift[i] = top[i] - 'A';
	
	int mod = 0;
	for(int i = 0; i < line.length(); i++)
	{
		if(line[i] >= 'A' && line[i] <= 'Z')
		{
			int num = ((line[i] - 'A') + shift[mod])%26;
			//cout << (char)(num + 'A') << ":" << num << ":" << (line[i] - 'A') << ":" << shift[mod] << ":" << mod <<":" << top.length() << endl;
			cout << (char)(num + 'A');

			mod++;
			if(mod == top.length()-1)
				mod = 0;
		}
	}
	cout << endl;
	
	return 0;
}

File: CompetitiveProgramming/CCC/CCC '04 J4 - Simple Encryption/2019-04-15/SimpleEncryption.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

public class SimpleEncryption
{
	public static void main(String [] args) throws IOException
	{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		String top = br.readLine();
		String line = br.readLine();
		
		int [] shift = new int [top.length()];
		for(int i = 0; i < top.length(); i++)
			shift[i] = top.charAt(i) - 'A';
		
		int mod = 0;
		for(int i = 0; i < line.length(); i++)
		{
			if(line.charAt(i) >= 'A' && line.charAt(i) <= 'Z')
			{
				int num = ((line.charAt(i) - 'A') + shift[mod])%26;
				System.out.print((char)(num + 'A'));

				mod++;
				if(mod == top.length())
					mod = 0;
			}
		}
		System.out.println();
	}
}